conn_linux_integration_test.go

```diff
@@ -0,0 +1,81 @@
+//+build integration,linux
+
+package netlink
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+
+	"reflect"
+)
+
+func TestLinuxNetlinkMulticast(t *testing.T) {
+	cfg := &Config{
+		Groups: 0x1, // RTMGRP_LINK
+	}
+
+	c, err := Dial(0, cfg) // dials NETLINK_ROUTE
+	if err != nil {
+		t.Fatalf("failed to dial: %v", err)
+	}
+
+	in := make(chan []Message)
+
+	// routine for receiving any messages
+	recv := func() {
+		data, err := c.Receive()
+		if err != nil {
+			panic(fmt.Sprintf("error in receive: %s", err))
+		}
+		in <- data
+	}
+
+	go recv()
+
+	ifName := "test0"
+
+	def := sudoIfCreate(t, ifName)
+	defer def()
+
+	timeout := time.After(5 * time.Second)
+	var data []Message
+	select {
+	case data = <-in:
+		break
+	case <-timeout:
+		t.Fatal("did not receive any messages after 5 seconds")
```

Comment:
`panic()` is nice in the case of timeouts because it'll give a stack trace showing where the goroutines are hung up.